src/ajax/util.cljc

```diff
@@ -26,6 +26,24 @@
                (.write ^String (to-str params))
                (.flush)))))
 
+(def successful-response-codes-set
+  "A set of successful response types derived from `goog.net.HttpStatus.isSuccess`."
+  ;; Factoid: Closure considers some 2XX status codes to *not* be successful, namely
+  ;; 205 Reset Content, 207 Multi Status & the unspecified 208+ range
+  #{200    ;; Ok
+    201    ;; Created
+    202    ;; Accepted
+    204    ;; No Content
+    206    ;; Partial Content
+    304    ;; Not Modified
+    ;; See https://github.com/google/closure-library/blob/f999480c4005641d284b86d82d0d5d0f05f3ffc8/closure/goog/net/httpstatus.js#L89-L94
+    1223}) ;; QUIRK_IE_NO_CONTENT
+
+
 (defn success? [status]
   "Indicates whether an HTTP status code is considered successful."
-  (<= 200 status 299))
+  ;; NOTE: cljs.ajax historically defined success as 200-299.
+  ;;       *DO NOT* remove (<= 200 status 299) as it will be
+  ;;       a breaking change
```

Comment:
Thanks for taking the time to contribute this PR, @avescodes 

As @JulianBirch mentions, the overall change to this function is a breaking one. Anyone already handling the fact that a `304` returns `false` for the previous version of this function will have their application broken upon upgrading.

With that in mind, I would suggest either (1) not making the version bump be patch-level, or better (2) accrete. Names like `successful?`, `success-status?`, `success-code?`, etc., could be used for a new function that relies solely on your `successful-response-codes-set`, leaving the current `success?` in place and marking it as deprecated in favor of your new function.

That approach would leave both functions simpler, allow you to introduce only a patch-level version bump, and would mean people using `success?` without issue don't have to be bothered to change their code until/unless they find the need.